src/Elastic.Apm/Logging/LogValuesFormatter.cs

```diff
@@ -16,13 +12,20 @@ namespace Elastic.Apm.Logging
 	/// </summary>
 	internal class LogValuesFormatter
 	{
-		private readonly string _scope;
 		private const string NullValue = "(null)";
 		private static readonly object[] EmptyArray = new object[0];
 		private static readonly char[] FormatDelimiters = { ',', ':' };
 		private readonly string _format;
+		private readonly bool _isArgumentNumberMismatch;
+
+		/// <summary>
+		/// Holds the list of placeholders that do not have corresponding values in the structured log.
+		/// </summary>
+		private readonly List<string> _mismatchedValueNames = new List<string>();
```

Comment:
IIUC `_isArgumentNumberMismatch` and `_mismatchedValueNames` are used only in the constructor so you can make them local variables. Also `_isArgumentNumberMismatch == true` == `mismatchedValueNames.Count > 0` so you can keep only `mismatchedValueNames`.

Reply:
Oh, right. In a previous version they were used outside, but not anymore - nice catch, changed it accordingly.